distro/exporter.go

```diff
@@ -62,26 +66,52 @@ func newOTLPExporter(c *exporterConfig) (trace.SpanExporter, error) {
 	return otlptracegrpc.New(context.Background(), opts...)
 }
 
-func otlpEndpoint(configured string) string {
+func otlpEndpoint(configured string) (string, bool) {
 	if configured != "" {
-		return configured
+		return configured, isLocalhost(configured)
 	}
 
 	// Allow the exporter to interpret these environment variables directly.
 	envs := []string{otelExporterOTLPEndpointKey, otelExporterOTLPTracesEndpointKey}
 	for _, env := range envs {
-		if _, ok := os.LookupEnv(env); ok {
-			return ""
+		if v, ok := os.LookupEnv(env); ok {
+			return "", isLocalhost(v)
```

Comment:
I think that it can be changed to:

```suggestion
			return "", false
```

As the OTel Go should be already setting secure/insecure

Reply:
We need to test the behavior when the local collector has TLS configured.

Reply:
Passing the TLS certificate pair you want to test with the `WithTLSConfig`. If the collector is running locally with TLS it will likely be using a non-host CA which means this options will be required regardless of this parsing. If they have updated the localhost CA, that can be accommodated by passing an empty config to that option as well.

Reply:
> If the collector is running locally with TLS it will likely be using a non-host CA which means this options will be required regardless of this parsing.


You are right 👍 

And what about "it will not connect to a remote collector that works without TLS." ? E.g. in the case of a K8s or docker-compose env like this

```
version: "3.9"
services:

  // some instrumented Go service
  server:
    // ...
    environment:
      - OTEL_SERVICE_NAME=server

      - OTEL_EXPORTER_OTLP_ENDPOINT=http://otel-collector:4317
    depends_on:
      - otel-collector

  otel-collector:
    image: quay.io/signalfx/splunk-otel-collector:0.48.0
    environment:
      - SPLUNK_ACCESS_TOKEN
      - SPLUNK_REALM
    ports:
      - "1777:1777"   # pprof extension
      - "8888:8888"   # Prometheus metrics exposed by the collector
      - "13133:13133" # health_check extension
      - "55679:55679" # zpages extension
      - "4317:4317"   # OTLP gRPC receiver
      - "4318:4318"   # OTLP HTTP receiver
 ```

moreover also similar problem would be when an OTel Collector would be used as a gateway/proxy and one would like to send the telemetry directly to it (without having a collector on localhost).

For sure this scenario should be supported: https://docs.splunk.com/Observability/gdi/get-data-in/application/go/instrumentation/instrument-go-application.html#deploy-the-go-instrumentation-in-kubernetes

Reply:
bump 😉

Reply:
I think I have addressed the issue based on our conversation. Please take another look, and unresolve if not.